core/src/main/java/com/opensymphony/xwork2/util/fs/StrutsJarURLConnection.java

```diff
@@ -18,6 +18,8 @@
  */
 package com.opensymphony.xwork2.util.fs;
 
+import sun.net.www.ParseUtil;
```

Comment:
`sun.net.www.ParseUtil` Sure?

Reply:
This was what my Oracle JDK 7 also imports in it's java.net.JarURLConnection class! I now researched and saw OpenJDK, Oracle JDK 8 and 9 also is same. I see sun.net.www.ParseUtil is inside rt.jar of these JDKs. It looks like opensymphony package name in Struts.

Reply:
Doesn't mean you should use it. ;) See [Replace sun.net.www.ParseUtil#decode with public API](https://bugs.openjdk.java.net/browse/JDK-8088916) and [sun.net.www.ParseUtil.decode() vs java.net.URLDecoder.decode()](https://stackoverflow.com/q/6869233/1700321)

Reply:
nice catch 👍 thanks! fixed.